src/main/scala/intellij/haskell/action/CreateHaskellFileAction.scala

```diff
@@ -51,6 +64,81 @@ class CreateHaskellFileAction extends CreateFileFromTemplateAction(CreateHaskell
     })
   }
 
+  private def createFileFromTemplate(pathItems: Option[List[String]], fileName: String, template: FileTemplate, fileDir: PsiDirectory): PsiFile = {
+    pathItems match {
+      case None => null
+      case Some(items) =>
+        // Adapted from super definition.
+        val mkdirs = new CreateFileAction.MkDirs(fileName, fileDir)
+        val name = mkdirs.newName
+        val dir = mkdirs.directory
+        val project = dir.getProject
+
+        val nameWithmodulePrefix = if (invalidPathItems(items) || items.isEmpty) {
+          name
+        } else {
+          items.mkString(".") + "." + name
+        }
+
+        // Patch props with custom property.
+        val props = FileTemplateManager.getInstance(project).getDefaultProperties()
+        props.setProperty("NAME", nameWithmodulePrefix)
+
+        val element = FileTemplateUtil.createFromTemplate(template, name, props, dir)
+        val psiFile = element.getContainingFile
+
+        try {
+          val virtualFile = Option(psiFile.getVirtualFile)
+
+          virtualFile.foreach(vFile => {
+            FileEditorManager.getInstance(project).openFile(vFile, true)
+            Option(getDefaultTemplateProperty).foreach(defaultTemplateProperty => {
+              PropertiesComponent.getInstance(project).setValue(defaultTemplateProperty, template.getName)
+            })
+          })
+        } catch {
+          case e: ParseException => Messages.showErrorDialog(project, "Error parsing Haskell Module template: " + e.getMessage, "Create File from Template");
+        }
+
+        psiFile
+    }
+  }
+
+  override def createFileFromTemplate(fileName: String, template: FileTemplate, fileDir: PsiDirectory): PsiFile = {
+    val path = fileDir.getVirtualFile.getPath
+    val pathItems = ProjectRootManager.getInstance(fileDir.getProject)
+      .getContentSourceRoots
+      .map(_.getPath)
+      .find(path.startsWith)
+      .map(s => if (s != path) {
+        path.replace(s + File.separator, "").split(File.separator).toList
```

Comment:
Oh no! This code is not working on Windows!
You bad bad!

Reply:
@ice1000 Sorry 😅

Reply:
🌘